core/processor.go

```diff
@@ -503,3 +515,53 @@ func (b *Processor) toLowLevelMessage(
 	}
 	return &withdrawalTx, nil
 }
+
+func (b *Processor) CheckByFilterOptions(botDelegatedWithdrawToEvent *L2ContractEvent, receipt *types.Receipt) error {
+	L2StandardBridgeBotAbi, _ := bindings2.L2StandardBridgeBotMetaData.GetAbi()
+	withdrawToEvent := bindings2.L2StandardBridgeBotWithdrawTo{}
+	indexedArgs := func(arguments abi.Arguments) abi.Arguments {
+		indexedArgs := abi.Arguments{}
+		for _, arg := range arguments {
+			if arg.Indexed {
+				indexedArgs = append(indexedArgs, arg)
+			}
+		}
+		return indexedArgs
+	}
+	err := abi.ParseTopics(&withdrawToEvent, indexedArgs(L2StandardBridgeBotAbi.Events["WithdrawTo"].Inputs), receipt.Logs[botDelegatedWithdrawToEvent.LogIndex].Topics[1:])
+	if err != nil {
+		return fmt.Errorf("parse indexed event arguments from log.topics of L2StandardBridgeBotWithdrawTo event, err: %v", err)
+	}
+
+	err = L2StandardBridgeBotAbi.UnpackIntoInterface(&withdrawToEvent, "WithdrawTo", receipt.Logs[botDelegatedWithdrawToEvent.LogIndex].Data)
+	if err != nil {
+		return fmt.Errorf("parse non-indexed event arguments from log.data of L2StandardBridgeBotWithdrawTo event, err: %v", err)
+	}
+
+	if len(b.cfg.L2StandardBridgeBot.WhitelistL2TokenList) > 0 {
+		isWhitelisted := false
+		for _, tokenAddress := range b.cfg.L2StandardBridgeBot.WhitelistL2TokenList {
+			if withdrawToEvent.L2Token.Cmp(common.HexToAddress(tokenAddress)) == 0 {
+				isWhitelisted = true
+				break
+			}
+		}
+		if !isWhitelisted {
+			return fmt.Errorf("filtered: token is not whitelisted, l2-token: %s", withdrawToEvent.L2Token)
+		}
+	}
+
+	if b.cfg.L2StandardBridgeBot.UpperMinGasLimit > 0 {
+		if withdrawToEvent.MinGasLimit > b.cfg.L2StandardBridgeBot.UpperMinGasLimit {
```

Comment:
Since 0 means no limit here, if you check with condition `withdrawToEvent.MinGasLimit > b.cfg.L2StandardBridgeBot.UpperMinGasLimit` here, you can never reject event `MinGasLimit == 1`.

- UpperMinGasLimit is 0, no limit so it will pass
- UpperMinGasLimit is 1, 1 == 1 so it will not return error

I suggest to check `>=` here to enable our check cover all cases we want.